appinventor/components/src/com/google/appinventor/components/scripts/ComponentProcessor.java

```diff
@@ -1389,6 +1717,247 @@ private Property executableElementToProperty(Element element, String componentIn
+        return optionListToHelperKey(optionList);
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Returns true if the given type implements OptionList. False otherwise.
+   * @param type the type to check if it implements OptionList.
+   * @return true if the given type implements OptionList. False otherwise.
+   */
+  private boolean isOptionList(TypeMirror type) {
+    if (type.getKind() == TypeKind.DECLARED) {
+      TypeElement elem = (TypeElement)((DeclaredType)type).asElement();
+      for (TypeMirror parent : elem.getInterfaces()) {
+        TypeElement parentElem = (TypeElement)((DeclaredType)parent).asElement();
+        if (parentElem.getSimpleName().toString().equals("OptionList")) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Returns the OptionList HelperKey associated with the given element.
+   * @param element the Element describing a class which implements the OptionList interface.
+   * @return the HelperKey associated with the given element.
+   */
+  private HelperKey optionListToHelperKey(Element optionList) {
+    String name = optionList.getSimpleName().toString();
+    // We haven't seen this type of option list before, so add it.
+    if (optionLists.get(name) == null) {
+      if (!tryAddOptionList(optionList)) {
+        // Couldn't add it for whatever reason.
+        return null;
+      }
+    }
+    // This helper key is storing info about an OptionList.
+    return new HelperKey(HelperType.OPTION_LIST, name);
+  }
+
+  /**
+   * Adds a new OptionList (based on the passed option list element) to the optionLists list.
+   * 
+   * @param optionElem The element representing the enum defining the options.
+   * @return Returns true if the Optionlist was successfully added. False otherwise.
+   */
+  private <E extends Enum<E>> boolean tryAddOptionList(Element optionElem) {
+    String className = optionElem.asType().toString();
+    String tagName = optionElem.getSimpleName().toString();
+    OptionList optionList = new OptionList(className, tagName);
+
+    // Get the class.
+    Class clazz = null;
+    try {
+      clazz = Class.forName(className);
+    } catch (ClassNotFoundException e) {
+      throw new IllegalArgumentException("OptionList Class: " + className + " is not available. " +
+        "Make sure that it is available to the compiler.");
+    }
+    if (clazz == null) {
+      return false;
+    }
+
+    // Get the getValue method.
+    java.lang.reflect.Method toValueMethod = null;
+    try {
+      toValueMethod = clazz.getDeclaredMethod("toUnderlyingValue", new Class[] {});
+    } catch (NoSuchMethodException e) {
+      throw new IllegalArgumentException("Class: " + className + " must have a toUnderlyingValue() " +
+          "method.");
+    }
+    if (toValueMethod == null) {
+      return false;
+    }
+
+    // Get the get method. This is not used, we just need to require it.
```

Comment:
Change "get method" to "fromUnderlyingValue method"

Reply:
[Done!](https://github.com/BeksOmega/appinventor-sources/pull/3/commits/730fe3f1a503c9902701a360d2e2783769c5dc12)